controller/controller.go

```diff
@@ -282,27 +303,32 @@ func UpdateExtensions(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	if jsonRequest {
+	// Set content type header
+	if omaha.IsJSONRequest(contentType) {
 		w.Header().Set("content-type", "application/json")
 	} else {
 		w.Header().Set("content-type", "application/xml")
 	}
 
 	w.WriteHeader(http.StatusOK)
-	updateResponse := updateRequest.FilterForUpdates(AllExtensionsMap)
 
-	if jsonRequest {
-		data, err = json.Marshal(&updateResponse)
-	} else {
-		data, err = xml.Marshal(&updateResponse)
+	// Use the generic FilterForUpdates function
+	updateResponse := extension.FilterForUpdates(updateRequest, AllExtensionsMap)
+
+	// Always use protocol 3.1 in responses regardless of input protocol
+	responseProtocolHandler, err := ProtocolFactory.CreateProtocol("3.1")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error creating response protocol handler: %v", err), http.StatusInternalServerError)
+		return
 	}
 
+	data, err := responseProtocolHandler.FormatResponse(updateResponse, false, contentType)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("Error in marshal %v", err), http.StatusInternalServerError)
+		http.Error(w, fmt.Sprintf("Error formatting response: %v", err), http.StatusInternalServerError)
 		return
 	}
 
-	if jsonRequest {
+	if omaha.IsJSONRequest(contentType) {
```

Comment:
Where can you see the logic here for the omaha class? The old behavior you can see at the top of `controller/controller.go` that just compares `contentType` against `"application/json"`. Just want to make sure nothing changes!

Reply:
Fair point - it sounds like something that can stay in `controller.go`.

Reply:
nm - I see in `omaha/utils.go` 😄 👍